### PR TITLE
Shuffles dynamic rulesets

### PR DESCRIPTION
**Pull request by EvilDragonfiend**
## About The Pull Request
Shuffles dynamic rulesets
because a first rulest will be always checked if it's acceptable, other rulesets will be likely to be chosen less. This isn't fair.
 

## Why It's Good For The Game
because this is real randomisation.

## Testing Photographs and Procedure

image

I don't know how to run dynamic simulation.

## Changelog
:cl:
code: dynamic rulesets are more fairly chosen rather than a specific ruleset is likely to be chosen more rather than actual randomisation.
/:cl:

### CODE REVIEW

File: code/game/gamemodes/dynamic/dynamic.dm [hunk 1]
```diff
@@ -538,7 +538,7 @@ GLOBAL_VAR_INIT(dynamic_forced_threat_level, -1)
 		log_game("DYNAMIC: Starting a round of forced extended.")
 		return TRUE
 	var/list/drafted_rules = list()
-	for (var/datum/dynamic_ruleset/roundstart/rule in roundstart_rules)
+	for (var/datum/dynamic_ruleset/roundstart/rule in shuffle(roundstart_rules))
```
Comment (PowerfulBacon) [inline code/game/gamemodes/dynamic/dynamic.dm:541]: This simply adds all of the rules into a list of drafted rules which uses a pickweight function to select the actual applied rule. This doesn't trigger the chosen ruleset, it simply filters them thus shuffling this list has no impact on the rules chosen. See line 556 just below for more info.

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]

File: code/game/gamemodes/dynamic/dynamic.dm (hunk 1 above)
Comment (EvilDragonfiend) [reply code/game/gamemodes/dynamic/dynamic.dm:541]: It appears the only line 541 doesn't need to be shuffled due to line 556, but others don't seem so.
Are you fine with other lines being suffled?

Comment (PowerfulBacon) [reply code/game/gamemodes/dynamic/dynamic.dm:541]: Where else is this used in a way that spawns rulesets?

File: code/game/gamemodes/dynamic/dynamic_midround_rolling.dm [hunk 2]
```diff
@@ -39,7 +39,7 @@
 	var/list/drafted_heavies = list()
 	var/list/drafted_lights = list()
 
-	for (var/datum/dynamic_ruleset/midround/ruleset in midround_rules)
+	for (var/datum/dynamic_ruleset/midround/ruleset in shuffle(midround_rules))
```
Comment (PowerfulBacon) [inline code/game/gamemodes/dynamic/dynamic_midround_rolling.dm:42]: This also only adds them to a list that gets picked randomly from using draft midround ruleset.
If these weren't random, then dynamic would be entirely deterministic and we would have the exact same gamemode every single round

**Comment by EvilDragonfiend** [comment]
oh, it already does that thing... I misunderstood the code.

**Pull request closed without merging**